File: src/game/scenes/MainScene.ts
import { Scene, GameObjects } from 'phaser';
import { Player } from '../gameObjects/Player';

export class MainScene extends Scene
{
    background: GameObjects.Image;
    logo: GameObjects.Image;
    title: GameObjects.Text;
    player: Player;

    constructor ()
    {
        super('MainScene');
    }

    create ()
    {
        // Create the player using the Player game object
        this.player = new Player(this, 448/2, 600);
    }

    update ()
    {
        if (this.player)
        {
            this.player.update();
        }
    }
}

File: csharp-game/Scenes/MainScene.cs
using System;

namespace PixelManGame.Scenes
{
    public class MainScene : Scene
    {
        public GameObject Background { get; set; }
        public GameObject Logo { get; set; }
        public GameObject Title { get; set; }
        public Player Player { get; set; }

        public MainScene() : base("MainScene")
        {
        }

        public override void Create()
        {
            // Create the player using the Player game object
            Player = new Player(this, 448 / 2, 600);
        }

        public override void Update(float time = 0, float delta = 16)
        {
            if (Player != null)
            {
                Player.Update(time, delta);
            }
        }
    }
}
